modemmanager: refactoring procd init script

feckert opened the pull request:
Maintainer: @aleksander0m @nickberry17 
Compile tested: not needed only script changes
Run tested: x86_64, APU3, latest OpenWrt master, started a modemmanger instance)

Description:
The way the init script is written now, we get a bad output when calling
the ubus service backend.

```
ubus call service list "{'verbose':true,'name':'modemmanager'}"
{
        "modemmanager": {
                "instances": {
                        "instance1": {
                                "running": true,
                                "pid": 20511,
                                "command": [
                                        "sh",
                                        "-c",
                                        ".
/usr/share/ModemManager/modemmanager.common;    \t
mkdir -m 0755 -p /var/run/modemmanager;          \t
mm_cleanup_interfaces;                            \t
( mm_report_events_from_cache ) >/dev/null 2>&1 & \t
/usr/sbin/ModemManager"
                                ],
                                "term_timeout": 5,
                                "respawn": {
                                        "threshold": 3600,
                                        "timeout": 5,
                                        "retry": 5
                                },
                                "pidfile":"/var/run/modemmanager/modemmanager.pid"
                        }
                }
        }
}
```

I also get the following output in the log that the PID file cannot be created:

`daemon.err procd: Failed to remove pidfile: :No such file or directory`

The changes in this commit fixes this issues

aleksander0m commented on net/modemmanager/files/modemmanager.init:
```diff
@@ -5,32 +5,24 @@ USE_PROCD=1
-	# procd-triggered respawns, which is why they're all included as instance command
-	#
+start_mm() {
+	. /usr/share/ModemManager/modemmanager.common
```
This is more or less what we had before, with the daemon call as only command in the procd instance. But having it that way ended up failing because mm_report_events_from_cache was not being called if the service was restarted by procd itself e.g. after a daemon crash (as explained in the comment that was also removed: "All these commands need to be executed on every MM start, even after procd-triggered respawns, which is why they're all included as instance command").

How are these changes avoiding that? If these changes prevent that problem, I'm all for them, but I'm not really sure if they do. Isn't procd supposed to re-run the "command" configured only on daemon crashes?

feckert replied:
If that's the case then I'd rather put it all in a wrapper/startup script `mm-service`?
Then it looks better right away.
We also could fix the two issues as mentioned (`ubus service` and `pdifle`)

aleksander0m replied:
Yes, that is probably the cleanest way forward

feckert replied:
@aleksander0m I have now outsourced the startup sequence to a wrapper script.

feckert commented:
@aleksander0m Can you look at that again.
I have now run shellcheck and fixes 4 things.

1. Disable SC2064 warning https://github.com/openwrt/packages/pull/17265/files#diff-2a022c1a8090bf6224c600ad07b0756c4f811b1b2b15ab7612a7bca07f3ee346R6

2. Split this command to fix SC2174
https://github.com/openwrt/packages/pull/17265/files#diff-2a022c1a8090bf6224c600ad07b0756c4f811b1b2b15ab7612a7bca07f3ee346R21

3. Made CHILD variable global https://github.com/openwrt/packages/pull/17265/files#diff-2a022c1a8090bf6224c600ad07b0756c4f811b1b2b15ab7612a7bca07f3ee346R26

4. Removed unused variables

nickberry17 commented:
ping @neheb LGTM
